### ISSUE TITLE
Cloud run django tutorial: gcloud builds submit fails

### ISSUE THREAD

**Issue by Theophylline** [GoogleCloudPlatform/python-docs-samples#6165]
## In which file did you encounter the issue?

python-docs-samples/run/django/

### Did you change the file? If so, how?

No, `git diff` returns nothing; I cloned this repo and did not make any changes. I added a  `.env` file as instructed in the [cloud run django tutorial ](https://cloud.google.com/python/django/run#first_deployment_of_the_application). 

```
DATABASE_URL=postgres:// ....
GS_BUCKET_NAME=<bucket name>
SECRET_KEY=<generated by a [django secret key generator](https://miniwebtool.com/django-secret-key-generator/)>
```

## Describe the issue

1. Created a new project on GCP
2. Followed the instructions in the [cloud run django tutorial ](https://cloud.google.com/python/django/run#first_deployment_of_the_application) _exactly_ just to get the hang of deploying a django app on cloud run. However, when I try to do an initial deployment with `gcloud builds submit --config cloudmigrate.yaml \
    --substitutions _INSTANCE_NAME=learn-db,_REGION=us-east1`, I get an error saying that the Cloud SQL instance does not exist:
![image](https://user-images.githubusercontent.com/19714287/122491323-c2d78f80-cfb1-11eb-82bc-74053f9e432c.png)

`gcloud sql instances list` for this project returns the following:

![image](https://user-images.githubusercontent.com/19714287/122491443-03cfa400-cfb2-11eb-89f2-ed6bc4fe7eba.png)

tl;dr: followed the Django Cloud Run docs exactly with the same code, but cannot deploy the application. Not sure what I did wrong. I've been stuck on this for 2 days, so I would appreciate any help. Thanks!

**Comment by dandhlee** [comment]
Hey @Theophylline, it seems like your instance is hosted in `us-central1-f` which is `us-central1` but you're running the command for `REGION=us-east1`. If you change that to

```
gcloud builds submit --config cloudmigrate.yaml \ --substitutions _INSTANCE_NAME=learn-db,_REGION=us-central-1
```
that should work as you hope! I'll leave this issue open, feel free to close it if this tweak helps you run further :)

**Comment by Theophylline** [comment]
Thanks, seems like the db instance can be found after changing the region to us-central1. But `python manage.py migrate` fails: 
```
Step #2 - "apply migrations": ---------- EXECUTE COMMAND ----------
Step #2 - "apply migrations": python manage.py migrate
Step #2 - "apply migrations": Traceback (most recent call last):
...
Step #2 - "apply migrations": psycopg2.OperationalError: could not connect to server: No such file or directory
Step #2 - "apply migrations":   Is the server running locally and accepting
Step #2 - "apply migrations":   connections on Unix domain socket "/cloudsql/project-name:us-east1:learn-db/.s.PGSQL.5432"?
```
The `DATABASE_URL` in my .env file is:
`DATABASE_URL = postgres://db-user:db-password@//cloudsql/project-name:us-central1:learn-db/db-name`. This is the only variable that I could think of that may cause this error because I didn't change anything else.
 
I followed the tutorial on the console so I don't have a history of commands, unfortunately.

**Comment by dandhlee** [comment]
It seems like the db instance might not be a Postgres database running. Could you check if your Cloud SQL instance is a PostgreSQL database? You can check by going to your GCP console, then Databases -> SQL, and check for the db instance you made.

If it is PostgreSQL database then could you make sure you have all the APIs enabled? Step 3 from "Before you begin" has the link for enabling the APIs or this link will also work: https://console.cloud.google.com/flows/enableapi?apiid=run.googleapis.com,sql-component.googleapis.com,sqladmin.googleapis.com,compute.googleapis.com,cloudbuild.googleapis.com,secretmanager.googleapis.com&_ga=2.116115654.1167906273.1623959282-1687404249.1620409723

Let me know if this doesn't work!

**Comment by Theophylline** [comment]
The instance is a postgresql database. `gcloud sql instances list` returns this:
```
NAME      DATABASE_VERSION  LOCATION       TIER              PRIMARY_ADDRESS  PRIVATE_ADDRESS  STATUS
learn-db  POSTGRES_13       us-central1-f  db-custom-1-3840  <ip address>     -                RUNNABLE
```

Here's my list of enabled APIs using `gcloud services list --enabled`:
```
NAME                              TITLE
bigquery.googleapis.com           BigQuery API
bigquerystorage.googleapis.com    BigQuery Storage API
cloudapis.googleapis.com          Google Cloud APIs
cloudbuild.googleapis.com         Cloud Build API
clouddebugger.googleapis.com      Cloud Debugger API
cloudtrace.googleapis.com         Cloud Trace API
compute.googleapis.com            Compute Engine API
containerregistry.googleapis.com  Container Registry API
datastore.googleapis.com          Cloud Datastore API
logging.googleapis.com            Cloud Logging API
monitoring.googleapis.com         Cloud Monitoring API
oslogin.googleapis.com            Cloud OS Login API
pubsub.googleapis.com             Cloud Pub/Sub API
run.googleapis.com                Cloud Run Admin API
secretmanager.googleapis.com      Secret Manager API
servicemanagement.googleapis.com  Service Management API
serviceusage.googleapis.com       Service Usage API
sql-component.googleapis.com      Cloud SQL
sqladmin.googleapis.com           Cloud SQL Admin API
storage-api.googleapis.com        Google Cloud Storage JSON API
storage-component.googleapis.com  Cloud Storage
storage.googleapis.com            Cloud Storage API
```
There might be some details that I missed when going through the tutorial, so maybe I should redo the tutorial but using the terminal this time? That way I can post the list of commands here. Is this a good tutorial to follow? https://codelabs.developers.google.com/codelabs/cloud-run-django/#0

**Comment by dandhlee** [comment]
Pinging our Serverless expert: @averikitsch, could you help take a look?

As per re-doing the tutorial, if you'd like to try again please do! We'll still note the friction you're having and try to improve the samples going forward.

I'm not too experienced with using Cloud Run, so I'll let others chimes in. Codelabs in general I think are good however :D

**Comment by Theophylline** [comment]
Following the Codelabs tutorial gave me a better understanding of where went wrong. It turns out that the `DATABASE_URL` in my local .env file is different from the .env file I uploaded to the secrets manager (there was a region mismatch). Correcting that allowed me to successfully deploy the app. Thanks for your help @dandhlee

**Comment by dandhlee** [comment]
Yay! I'm glad things got resolved :)

**Comment by Theophylline** [comment]
Sorry to reopen this issue. While deploying my own app (not the codelab example) and creating a superuser as a data migration as described [here](https://cloud.google.com/python/django/run#superuser_creation_with_data_migrations), I cannot login with the correct username and password from django admin. I get `Please enter the correct username and password for a staff account. Note that both fields may be case-sensitive.` on the login page.

The tables were created properly though; I confirmed by connecting to the cloud sql instance and listed the tables in my db, so `python manage.py migrate`  in the `cloudmigrate.yaml` must have worked...? Instead of going back to find out what went wrong with the deployment steps, I was wondering if there's an easy way to add a superuser after the app is deployed? Thanks
